### PR TITLE
feature: Add custom exceptions

### PR DESCRIPTION
**Pull request by makcimmushka**
Add file operation exceptions for all methods in FileManager

### CODE REVIEW

File: second-semester/Filesystem/src/main/java/ua/knu/cli/command/Command.java [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 import lombok.val;
+import ua.knu.exceptions.FileOperationException;
 import ua.knu.util.Constants;
 
 public abstract class Command {
 
     public abstract boolean canProcess(String command);
 
-    public abstract void process(String command);
+    public abstract void process(String command) throws FileOperationException;
```
Comment (Vlad-Yakimenko) [inline second-semester/Filesystem/src/main/java/ua/knu/cli/command/Command.java]: Why abstract command class throws specific for file interaction exception, this breaks command abstraction

File: second-semester/Filesystem/src/main/java/ua/knu/cli/command/OpenCommand.java [hunk 2]
```diff
@@ -21,7 +22,7 @@ public boolean canProcess(String command) {
     }
 
     @Override
-    public void process(String command) {
+    public void process(String command) throws FileOperationException {
```
Comment (Vlad-Yakimenko) [inline second-semester/Filesystem/src/main/java/ua/knu/cli/command/OpenCommand.java]: ~

File: second-semester/Filesystem/src/main/java/ua/knu/cli/command/Command.java (hunk 1 above)
Comment (Vlad-Yakimenko) [reply second-semester/Filesystem/src/main/java/ua/knu/cli/command/Command.java]: Use SneakyThrows instead

Comment (makcimmushka) [reply second-semester/Filesystem/src/main/java/ua/knu/cli/command/Command.java]: OK

File: second-semester/Filesystem/src/main/java/ua/knu/cli/command/OpenCommand.java (hunk 2 above)
Comment (makcimmushka) [reply second-semester/Filesystem/src/main/java/ua/knu/cli/command/OpenCommand.java:25]: See answer above

**Comment by Vlad-Yakimenko** [comment]
LGTM!

**Pull request merged**